File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/model/v2/group/Group.kt
package de.thm.ii.fbs.model.v2.group

import de.thm.ii.fbs.model.v2.security.User
import javax.persistence.*

@Entity
@Table(name = "`group`")
class Group(
    @Column(nullable = false)
    var courseId: Int,
    @Column(nullable = false)
    var name: String,
    @Column(nullable = false)
    var membership: Int,
    @Column(nullable = false)
    var visible: Boolean,
    @OneToMany(mappedBy = "groupId")
    var users: List<UserGroup>,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "group_id")
    var id: Int? = null
)

File: modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/model/Group.scala
package de.thm.ii.fbs.model

/**
  * A group
  * @param id The id of the group
  * @param courseId course to which the group belongs
  * @param name Name of the group
  * @param membership The max number of members
  * @param visible The visibility of the group, false = invisible
  */

case class Group(id: Int, courseId: Int, name: String, membership: Int, visible: Boolean = true)
